### ISSUE TITLE
`opam pin remove` behaviour

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/opam#3630]
I think this used to work in the v2 betas but it seems it no longer. 

When you release you usually have the package you release as a `pin` on the current version `X`. You submit to the opam repo version `X+1`. Then when you `opam update` and have the update `X+1` from the OCaml opam repository, you want to remove the pin and recompile with `X+1`. The current behaviour when you `opam pin remove` is to recompile with `X` which is annoying if your package has many deps, since you have to follow with a recompile with `X+1`.

**Comment by rjbou** [comment]
It's only if version `X` is present in the repo. It has been designed that way, since [1.2.1](https://github.com/ocaml/opam/commit/c90549334b6fd2e9e5bba17ed9ef8c24bba6c678#diff-efc403d1b034c3c1403258dac55c0b30R1636).

**Comment by dbuenzli** [comment]
I don't understand your answer.

**Comment by dbuenzli** [comment]
If that's not clear what I'm asking is that when you remove a dev pin it tries to install the latest version of the package not the version that was there when you pinned.

**Comment by rjbou** [comment]
It's clear :) What I wanted to say (with few words) is that there is two processes:
* unpin package version `X`, and version `X` exists in opam repository → opam reinstall package version `X`
* unpin package version `X`, and version `X` absent from opam repository → opam reinstall package to its last version (`X+1` here)
It's a design choice, that can be discussed ofc.

**Comment by dbuenzli** [comment]
Thanks. There are a few things that are related to this which are 

https://github.com/ocaml/opam/issues/3202
https://github.com/ocaml/opam/issues/2932

**Comment by dbuenzli** [comment]
IIRC there's actually a workaround -- the problem is that I usually forget about it and blindly issue an `opam pin remove cmdliner` and then I'm very sad. I think what you have to do is `opam pin remove PKG -n` and then `opam upgrade PKG`. 

I was always a bit uncomfortable with it because you basically you go through an inconsistent switch state. 

But if people consider that's good enough maybe it's not worth changing/complexifying things and this can be closed. (I let you do if you think so)

**Comment by rjbou** [comment]
After discussion with @AltGr, it makes sense to remove that specific behaviour, in order to always have the last repo version of the newly unpinned package.
